### ISSUE TITLE
Sort-AppFoldersByDependencies can't handle twice the same app in the baseFolder

### ISSUE THREAD

**Issue by Spycoclown** [microsoft/navcontainerhelper#1507]
We have a system that downloads the needed dependencies from the artifacts, in the artifacts the apps also have their needed dependencies so it can happen that twice the same app is inside the folder structure.

**Scripts used to create container and cause the issue**
```
 Sort-AppFoldersByDependencies -appFolders $appList.Split(',') -baseFolder $baseFolder -WarningAction SilentlyContinue | ForEach-Object {
```

**Full output of scripts**
```
BcContainerHelper version 1.0.14-dev

##[error]ForEach-Object : Item has already been added. Key in dictionary: '6a6d936f-1c1f-44c8-b3b5-06bcb744a2c2'  Key being 
added: '6a6d936f-1c1f-44c8-b3b5-06bcb744a2c2'
At C:\Program 
Files\WindowsPowerShell\Modules\BcContainerHelper\1.0.14\AppHandling\Sort-AppFoldersByDependencies.ps1:36 char:19
+     $appFolders | ForEach-Object {
+                   ~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (:) [ForEach-Object], ArgumentException
    + FullyQualifiedErrorId : System.ArgumentException,Microsoft.PowerShell.Commands.ForEachObjectCommand
##[error]PowerShell exited with code '1'.
```

**Comment by freddydk** [comment]
Shipped in 1.0.15-preview275

**Comment by Spycoclown** [comment]
Hello Freddy,

is just tested your fix, unfortunately, you fixed only AppFilesByDependencies and not AppFoldersByDependencies.

```
BcContainerHelper version 1.0.15
##[error]ForEach-Object : Item has already been added. Key in dictionary: '6a6d936f-1c1f-44c8-b3b5-06bcb744a2c2'  Key being 
added: '6a6d936f-1c1f-44c8-b3b5-06bcb744a2c2'
At C:\Program 
Files\WindowsPowerShell\Modules\BcContainerHelper\1.0.15\AppHandling\Sort-AppFoldersByDependencies.ps1:36 char:19
+     $appFolders | ForEach-Object {
+                   ~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (:) [ForEach-Object], ArgumentException
    + FullyQualifiedErrorId : System.ArgumentException,Microsoft.PowerShell.Commands.ForEachObjectCommand
##[error]PowerShell exited with code '1'.
```
